src/protocols/secure_channel/PairingSession.cpp

```diff
@@ -18,9 +18,23 @@
 
 #include <protocols/secure_channel/PairingSession.h>
 
+#include <app/DataModelRevision.h>
+#include <app/InteractionModelRevision.h>
+#include <lib/core/CHIPConfig.h>
 #include <lib/core/TLVTypes.h>
 #include <lib/support/SafeInt.h>
 
+#define IncrementReaderExitOnEndOfTlv(tlvReader)                                                                                   \
```

Comment:
since this macro contains a `return` I think it should look more like a macro instead of just afunctioncall.
How about `INCREMENT_READER_EXIT_ON_END_OF_TLV` ? This is different than ReturnOnError/VerifyOrExit, however with those I was a bit more familiar. This seemed like a function call the first time I read through the code.

Reply:
Also, it's not clear what "Exit" means here.

This would be clearer as "NextElementOrExitContainerAndReturn" or something (I'm agnostic to the casing).

Reply:
Or perhaps this is just doing too much (and hence we are having such a hard time with it), and we should in fact not have a macro that does both of these things and instead restructure some of the other code to make it less repetitive in a different way....

Reply:
I still need to do this, and will finish this off tomorrow, but I have to run for today